the-collab-lab/tcl-43-smart-shopping-list
# La ng add react router dom

## Description
This PR adds 2 new components: one for the "shopping list" view and one for the "add item" view.

The NavLink element from React Router Dom allows us to detect which route is currently active and provides an isActive property which contains a boolean value. We've used the isActive property to provide conditional styling to the "add item" and "shopping list" links according to which link is currently active.

Documentation for React Router Dom can be found [here] (https://reactrouter.com/docs/en/v6/getting-started/overview).
A straightforward explanation of the NavLink element with example code can be found [here] (https://ultimatecourses.com/blog/active-navlink-inline-styles-with-react-router).



## Related Issue
closes #2



## Acceptance Criteria

- [x] Whichever view is selected, the corresponding link displays in bold text


## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
| ✓   | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before
started from scratch

### After

<img width="866" alt="shopApp1" src="https://user-images.githubusercontent.com/60862882/162579693-c7db5a9f-59b6-43b2-a2ae-27b3ccbffb75.png">
<img width="866" alt="shopApp2" src="https://user-images.githubusercontent.com/60862882/162579699-d258f4d7-76a8-4106-898a-7f4c11a6d7f6.png">





## Testing Steps / QA Criteria

- From your terminal, pull down this branch with git pull origin la-ng-add-react-router-dom and check that branch out with git checkout la-ng-add-react-router-dom
- Then npm to install the newly added dependencies locally and npm start to launch the app.
- Click the links to "shopping list" and "add item" at the bottom of the page. Make sure the URL and link styling changes according to the current path.
